_docs-sources/iac/usage/composing-your-own-service.md

```diff
@@ -0,0 +1,193 @@
+# Composing your own Service
+
+We offer a collection of [services](../overview/services.md#what-is-a-service) that piece together individual [modules](../overview/modules.md#what-is-a-module) for specific use cases such as EKS clusters and VPCs with public and private subnets. While we strive to make our service catalog as a complete as possible, you may need to create your own service to suit a specific use case or need for your company.
+
+In this guide, you will learn how to create a service that provisions a simple API using the [AWS Lambda Function](../../reference/modules/terraform-aws-lambda/lambda/) and [API Gateway](../../reference/modules/terraform-aws-lambda/lambda-http-api-gateway/) modules from the Gruntwork Infrastructure as Code (IaC) Library.
+
+## Prerequisites
+
+- An AWS account with permissions to create the necessary resources
+- An [AWS Identity and Access Management](https://aws.amazon.com/iam/) (IAM) user or role with permissions to create AWS IAM roles, Lambda functions, and API Gateways
+- [AWS Command Line Interface](https://aws.amazon.com/cli/) (AWS CLI) installed on your local machine
+- [Terraform](https://www.terraform.io) installed on your local machine
+
+## Creating the service
+
+In this section we’ll define a service that provisions an AWS Lambda Function and HTTP API Gateway, with a single proxy route on the API Gateway pointing to the Lambda. This service exposes a simple set of inputs for configuring the Lambda code configuration and name of the provisioned resources.
+
+### Create the basic file structure
+
+First, create the basic file structure that will contain the service definition. We’ll create three files — main.tf which will contain the resource definitions, variables.tf, which specifies the possible inputs to the module, and outputs.tf, which specifies the values that can be used to pass references to attributes from the resources in the module.
+
+```bash
+mkdir -p gw_service_guide/serverless-api/
+touch gw_service_guide/serverless-api/main.tf
+touch gw_service_guide/serverless-api/variables.tf
+touch gw_service_guide/serverless-api/outputs.tf
+```
+
+### Defining the service
+
+Next, define the module blocks for the AWS Lambda function and HTTP API Gateway. For the Lambda, we will use the [`Lambda function` module](../../reference/modules/terraform-aws-lambda/lambda/), for the HTTP API Gateway we will use the [`HTTP API Gateway` module](../../reference/modules/terraform-aws-lambda/lambda-http-api-gateway/).
```

Comment:
It weirds me out that only part of the link is a code block... not entirely sure what the fix is... should we just make them links and not code blocks?

Reply:
Yeah, good call. I was thinking of a conversation w/ Eben and Eugene about how we stylize these, but I misremembered the implementation (we only do the code block for links with the repo name).

Reply:
Resolved in https://github.com/gruntwork-io/docs/pull/846/commits/822ebcde2f9c781a89b82f0786e8fa0da7bc9728